a3m/executeOrRunSubProcess.py

```diff
@@ -131,25 +131,24 @@ def launchSubProcess(
 def createAndRunScript(
     text, stdIn="", printing=False, arguments=[], env_updates={}, capture_output=True
 ):
-    # Output the text to a /tmp/ file
-    scriptPath = "/tmp/" + uuid.uuid4().__str__()
-    FILE = os.open(scriptPath, os.O_WRONLY | os.O_CREAT, 0o770)
-    os.write(FILE, text.encode("utf8"))
-    os.close(FILE)
-    cmd = [scriptPath]
-    cmd.extend(arguments)
-
-    # Run it
-    ret = launchSubProcess(
-        cmd,
-        stdIn="",
-        printing=printing,
-        env_updates=env_updates,
-        capture_output=capture_output,
-    )
-
-    # Remove the temp file
-    os.remove(scriptPath)
+    with tempfile.NamedTemporaryFile(
+        dir=tempfile.gettempdir(), encoding="utf-8", mode="wt", delete=False
+    ) as FILE:
```

Comment:
Can we take this chance to choose a better name for the temporary file? E.g. `tmpfile` or `temp`.

Reply:
As it is the file name by default will be like: `/tmp/tmpupbhxx6r`  ('tmp' with a unique hash appended for uniqueness).

The command allows for a prefix and/or suffix to be specified instead of the default prefix `tmp`. e.g.: `/tmp/PREFIXxvd7rgqfSUFFIX`

Reply:
Sorry, I meant the name of the variable? `FILE`

Reply:
I see - yes, this is a good time to update that.